include/nbl/asset/utils/IMeshPacker.h

```diff
@@ -297,30 +297,104 @@ class IMeshPacker : public IMeshPackerBase
         Triangle* triangleArrayBegin = triangleBatches.triangles.data();
         Triangle* triangleArrayEnd = triangleArrayBegin + triangleBatches.triangles.size();
         const uint32_t triangleCnt = triangleBatches.triangles.size();
-        //for (uint32_t i = 0u; i < batchCnt; i++)
-        //    triangleBatches.ranges.push_back(triangleArrayBegin + (m_minTriangleCountPerMDIData * i)); //is calcBatchCountBound correct?
-        //triangleBatches.ranges.push_back(triangleArrayEnd);
 
-        std::random_device rd;
-        std::mt19937 mt(rd());
-        std::uniform_int_distribution<uint16_t> dist(m_minTriangleCountPerMDIData, m_maxTriangleCountPerMDIData);
-
-        triangleBatches.ranges.push_back(triangleArrayBegin);
-        for(uint32_t i = 1u, trianglesAddedToBatchesCnt = 0u; ; i++)
+        //aabb batch division
         {
-            const uint16_t randomBatchSize = dist(mt);
-            trianglesAddedToBatchesCnt += randomBatchSize;
-
-            //std::cout << randomBatchSize << std::endl;
-
-            if (trianglesAddedToBatchesCnt >= triangleCnt)
+            triangleBatches.ranges.push_back(triangleArrayBegin);
+            for (uint32_t i = 1u; ; i++)
             {
-                //std::cout << std::endl << trianglesAddedToBatchesCnt << std::endl;
-                triangleBatches.ranges.push_back(triangleArrayEnd);
-                break;
+                Triangle* batchBegin = *(triangleBatches.ranges.end() - 1u);
+                Triangle* batchEnd = batchBegin + m_minTriangleCountPerMDIData;
+                
+                if (batchEnd >= triangleArrayEnd)
+                {
+                    triangleBatches.ranges.push_back(triangleArrayEnd);
+                    break;
+                }
+
+                if (m_minTriangleCountPerMDIData == m_maxTriangleCountPerMDIData)
+                {
+                    triangleBatches.ranges.push_back(batchEnd);
+                    continue;
+                }
+
+                //find min and max edge
+                core::vector3df min(std::numeric_limits<float>::max());
+                core::vector3df max(std::numeric_limits<float>::min());
```

Comment:
you want `-std::numeric_limits<float>::max()` here

the `min()` returns `FLT_MIN` which is actually a small number close to 0